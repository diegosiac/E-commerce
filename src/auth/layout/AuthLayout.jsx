import { Grid, Typography } from '@mui/material'
import PropTypes from 'prop-types'

export const AuthLayout = ({ children, title }) => {
  return (
    <Grid
      item
      component='main'
      bgcolor='grey.50'
      className='max-w-[90%] p-6 sm:w-[450px] rounded-lg my-12'
    >
      <Typography variant='h2' component='h1' className='mb-2'>{title}</Typography>
      {children}
    </Grid>
  )
}

AuthLayout.defaultProps = {
  title: ''
}

AuthLayout.propTypes = {
  children: PropTypes.node,
  title: PropTypes.string
}
