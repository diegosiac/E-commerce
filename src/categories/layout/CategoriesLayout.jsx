import { Box, Grid, Typography } from '@mui/material'
import Proptypes from 'prop-types'
import { CardContend } from '../components/CardContend'
import hardware from '../../assets/imgs/backgrounds/hardware.webp'

export const CategoriesLayout = ({
  children,
  title,
  disableGrid,
  urlSecondary,
  titlePrimary,
  titleSecondary,
  titleThird,
  titleFourth,
  descriptionPrimary,
  descriptionSecondary,
  descriptionThird,
  descriptionFourth
}) => {
  return (
    <Grid
      container
      className='justify-center flex-col items-center mb-16'
    >

      <Grid
        container
        className='w-11/12 mt-5 mb-14'
      >
        <Box className='flex mb-8'>
          <Box className='w-2 h-full bg-[#3483fa] mr-2' />
          <Typography variant='h3' component='h2'>{title}</Typography>
        </Box>

        <Grid
          container
          className='justify-evenly gap-6'
        >
          {children}
        </Grid>

      </Grid>

      {
        !disableGrid &&
          <Box
            className='grid grid-cols-2 gap-4 w-11/12 md:grid-cols-4'
          >

            <CardContend
              classImg='object-cover h-[200px]'
              urlImg={hardware}
              title={titlePrimary}
              description={descriptionPrimary}
              className='col-span-2 md:row-span-2'
            />

            <CardContend
              classImg='object-contain w-32'
              urlImg={urlSecondary}
              title={titleSecondary}
              description={descriptionSecondary}
              className='flex col-span-2'
            />

            <CardContend
              title={titleThird}
              description={descriptionThird}
            />

            <CardContend
              title={titleFourth}
              description={descriptionFourth}
            />

          </Box>
      }

    </Grid>
  )
}

CategoriesLayout.proptypes = {
  children: Proptypes.node.isRequired,
  title: Proptypes.string.isRequired,
  disableGrid: Proptypes.bool,
  urlSecondary: Proptypes.string,
  titlePrimary: Proptypes.string,
  titleSecondary: Proptypes.string,
  titleThird: Proptypes.string,
  titleFourth: Proptypes.string,
  descriptionPrimary: Proptypes.string,
  descriptionSecondary: Proptypes.string,
  descriptionThird: Proptypes.string,
  descriptionFourth: Proptypes.string
}

CategoriesLayout.defaultProps = {
  disableGrid: false
}
