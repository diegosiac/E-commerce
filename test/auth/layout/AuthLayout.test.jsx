import { render, screen } from '@testing-library/react'
import { AuthLayout } from '../../../src/auth/layout/AuthLayout'
import { MemoryRouter } from 'react-router-dom'

const titleTest = 'title test'
const childrenTest = 'contend test'

describe('Testing on <AuthLayout/>', () => {
  test('should render the component correctly', () => {
    render(
      <MemoryRouter initialEntries={['/']}>
        <AuthLayout title={titleTest}>
          {childrenTest}
        </AuthLayout>
      </MemoryRouter>
    )

    expect(screen.getByText(titleTest)).toBeTruthy()

    expect(screen.getByText(childrenTest)).toBeTruthy()
  })

  test('must match with the Snapshot', () => {
    const tree = render(
      <MemoryRouter initialEntries={['/']}>
        <AuthLayout title={titleTest}>
          {childrenTest}
        </AuthLayout>
      </MemoryRouter>
    )

    expect(tree).toMatchSnapshot()
  })
})
