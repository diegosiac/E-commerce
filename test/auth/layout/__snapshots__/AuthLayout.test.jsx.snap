// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on <AuthLayout/> must match with the Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="MuiGrid-root MuiGrid-item max-w-[90%] p-6 sm:w-[450px] rounded-lg my-12 css-hp5ulv-MuiGrid-root"
      >
        <h1
          class="MuiTypography-root MuiTypography-h2 mb-2 css-1sra7t5-MuiTypography-root"
        >
          title test
        </h1>
        contend test
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="MuiGrid-root MuiGrid-item max-w-[90%] p-6 sm:w-[450px] rounded-lg my-12 css-hp5ulv-MuiGrid-root"
    >
      <h1
        class="MuiTypography-root MuiTypography-h2 mb-2 css-1sra7t5-MuiTypography-root"
      >
        title test
      </h1>
      contend test
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
